### PR TITLE
A batch of small site improvements.

### PR DESCRIPTION
**Pull request by geeknoid**
- We no longer have a redirect from the top-nav Blog link to the
current latest blog post, which looks nicer.

- Added a link to our team drive to the community page.

- Upgrade a few icons from PNG to SVG

- Reintroduce content compression which I accidentally left turned off previously.

- Reintroduce the RSS icon on blog posts which disappeared in last week's great blog
page redesign.

- Update doc contribution guide to cover blog posts and setup pages.

- Add templates for setup topics and blog posts.

Staging site: https://geeknoid.github.io/istio.github.io/

### CODE REVIEW

File: _data/istio.yml [hunk 1]
```diff
@@ -1 +1,2 @@
 version: 0.3.0
+latest_blog_post: 2017/mixer-spof-myth.html
```
Comment (ldemailly) [inline _data/istio.yml]: anyway to not need this to be manually updated / have it be the latest automatically?

File: _data/istio.yml [hunk 2]
```diff
@@ -1 +1,2 @@
 version: 0.3.0
```
Comment (ldemailly) [inline _data/istio.yml]: change to 0.4.0 while touching this file

File: _blog/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+layout: compress
+---
+{% include home.html %}
+
+<!DOCTYPE html>
+<html lang="en-US">
+  <meta charset="utf-8">
+  <title>Redirecting…</title>
+  <link rel="canonical" href="{{home}}/blog/posts/{{site.data.istio.latest_blog_post}}">
+  <meta http-equiv="refresh" content="0; url={{home}}/blog/posts/{{site.data.istio.latest_blog_post}}">
```
Comment (ldemailly) [inline _blog/index.html]: that’s still a hand crafted redirec

Comment (geeknoid) [reply _blog/index.html]: Yes, if you go to istio.io/blog, you'll get a redirect to the latest post. But if you click on the link in the nav bar, then you no longer get a redirect.

File: _data/istio.yml (hunk 2 above)
Comment (geeknoid) [reply _data/istio.yml]: Done

File: _data/istio.yml (hunk 1 above)
Comment (geeknoid) [reply _data/istio.yml]: I don't know how, given our current workflow. I at least managed to centralize things so there's only one place to update.

Maybe there'd be a way to look at the collation order on the blog post page and always target the top-most page. I'll give that a whirl.

**Pull request merged**